tests/test_workspace.py

```
@@ -675,3 +676,76 @@ def test_reopen_is_consistent_with_save_staleness(self, temp_docx):
         # And the documented recovery (drop the workspace) works.
         Workspace.delete(temp_docx)
         Workspace(temp_docx, author="Test").close()
+
+
+class TestDirtyWorkspace:
+    """Tests for the dirty flag guarding stale-workspace adoption (issue #31)."""
+
+    def _meta_on_disk(self, ws):
```

Ruff (0.15.21)

[warning] Missing return type annotation for private function `_meta_on_disk`

(ANN202)